### PR TITLE
Minor patch

### PR DESCRIPTION
**Pull request by sheungon**
### Description of code - what bug does this fix / what feature does this add?
Added sandbox endpoints.
Fixed few minor bugs.
Added AscendEX Futures.

- [x] - Tested
- [x] - Changelog updated
- [ ] - Tests run and pass
- [ ] - Flake8 run and all errors/warnings resolved
- [x] - Contributors file updated (optional)

### CODE REVIEW

File: cryptofeed/exchanges/ascendex.py [hunk 1]
```diff
@@ -24,12 +24,25 @@
...
         L2_BOOK: 'depth:',
         TRADES: 'trades:',
     }
+    # Docs, https://ascendex.github.io/ascendex-pro-api/#websocket-authentication
+    # noinspection PyTypeChecker
+    websocket_endpoints = [WebsocketEndpoint(
+        'wss://ascendex.com/1/api/pro/v1/stream',
+        channel_filter=(
+            websocket_channels[L2_BOOK],
+            websocket_channels[TRADES],
```
Comment (bmoscon) [inline cryptofeed/exchanges/ascendex.py:42]: why did you add a channel filter here? Seems like that would only be needed if you had added ascendex futures to this class as a second websocket endpoint.

File: cryptofeed/exchanges/ascendex_futures.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    websocket_channels = {
+        **AscendEX.websocket_channels,
+    }
+    # Docs, https://ascendex.github.io/ascendex-futures-pro-api-v2/#how-to-connect
+    # noinspection PyTypeChecker
+    websocket_endpoints = [WebsocketEndpoint(
+        'wss://ascendex.com:443/api/pro/v2/stream',
+        channel_filter=(
+            websocket_channels[L2_BOOK],
+            websocket_channels[TRADES],
```
Comment (bmoscon) [inline cryptofeed/exchanges/ascendex_futures.py:26]: why channel filters?

File: cryptofeed/exchanges/bitmex.py [hunk 3]
```diff
@@ -804,3 +810,21 @@ async def _authenticate(self, conn: AsyncConnection):
             msg = f'GET/realtime{expires}'.encode('utf-8')
             signature = hmac.new(self.key_secret.encode('utf-8'), msg, digestmod=hashlib.sha256).hexdigest()
             await conn.write(json.dumps({'op': 'authKeyExpires', 'args': [self.key_id, expires, signature]}))
+
+    async def book_callback(self, book_type: str, book: OrderBook, receipt_timestamp: float, timestamp=None, raw=None,
```
Comment (bmoscon) [inline cryptofeed/exchanges/bitmex.py:814]: I dont understand the purpose of overloading this method in bitmex. if you need to do something special for timestamps, why not just do that in the method that parses the book?

File: cryptofeed/exchanges/kraken_futures.py [hunk 4]
```diff
@@ -255,3 +262,27 @@ async def message_handler(self, msg: str, conn, timestamp: float):
                 await self._book(msg, pair, timestamp)
             else:
                 LOG.warning("%s: Invalid message type %s", self.id, msg)
+
+    async def book_callback(
```
Comment (bmoscon) [inline cryptofeed/exchanges/kraken_futures.py:266]: again, why?

**Review by bmoscon — CHANGES_REQUESTED** [review]
see the comments in the code. also, i know its not popular, but I'd rather see longer lines than 1 line turned into like 5 lines

File: cryptofeed/exchanges/ascendex_futures.py (hunk 2 above)
Comment (sheungon) [reply cryptofeed/exchanges/ascendex_futures.py:26]: As they have private endpoint for private stream, it is for futures private channels support.

See, https://ascendex.github.io/ascendex-futures-pro-api-v2/#how-to-connect

File: cryptofeed/exchanges/ascendex.py (hunk 1 above)
Comment (sheungon) [reply cryptofeed/exchanges/ascendex.py:42]: As they have private endpoint for private stream, it is for private channels support in the future.

See, https://ascendex.github.io/ascendex-pro-api/#websocket-request

File: cryptofeed/exchanges/bitmex.py (hunk 3 above)
Comment (sheungon) [reply cryptofeed/exchanges/bitmex.py:814]: Sorry, you're right. It is more direct to have this change in book parser method, `_book`.
I did this as I am having this fix in my project. Overriding `book_callback` will be a shorter solution.

Let me update this change.

File: cryptofeed/exchanges/kraken_futures.py (hunk 4 above)
Comment (sheungon) [reply cryptofeed/exchanges/kraken_futures.py:266]: Let me update this change too

**Review by bmoscon — APPROVED** [review]
